util_for_test.go

```diff
@@ -0,0 +1,329 @@
+package surrealdb
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	rawslog "log/slog"
+	"net"
+	"os/exec"
+	"reflect"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/surrealdb/surrealdb.go/pkg/conn/gorilla"
+	"github.com/surrealdb/surrealdb.go/pkg/logger/slog"
+)
+
+var (
+	testHasSurrealCLI     bool
+	checkSurrealCLIOnce   sync.Once
+	DelayAfterServerStart time.Duration = 300 * time.Millisecond
+	DelayBeforeServerExit time.Duration = 300 * time.Millisecond
+)
+
+func Test_NewTestSurrealDB_Simple(t *testing.T) {
+	// Simple illustration of how NewTestSurrealDB can be used.
+
+	// Tests can run in parallel.
+	t.Parallel()
+
+	// NewTestSurrealDB returns endpoint (random open port used), DB instance
+	// (which is wrapped in DBForTest to provide extra methods), and a func to
+	// shutdown the SurrealDB server.
+	endpoint, db, close := NewTestSurrealDB(t)
+	defer close()
+	_ = endpoint // endpoint could be useful for some low level testing.
+
+	// DBForTest.Prepare is to run any SurrealQL for test prep. This would fail
+	// early if the provided string results in an error from SurrealDB server.
+	db.Prepare(t, "CREATE user:x SET name = 'x'")
+
+	// More interactions can simply use db instance from here on.
+}
+
+func Test_TestSurrealDBParallelSimple(t *testing.T) {
```

Comment:
I named the main func `NewTestSurrealDB`, so I thought it would be clearer to tell what I'm testing is `TestSurrealDB`. Perhaps it may be better to name the function to something like `NewDummySurrealDB` or `NewSurraelDBForTest` to avoid the stutter?